src/storage/storage-engine.cpp

```diff
@@ -269,28 +269,28 @@ ndn::ptr_lib::shared_ptr<Data> StorageEngineImpl::get(const Name &dataName)
                                       &dataString);
     if (s.ok())
     {
-        ndn::ptr_lib::shared_ptr<Data> data = ndn::ptr_lib::make_shared<Data>();
+        std::shared_ptr<Data> data = std::make_shared<Data>();
         data->wireDecode((const uint8_t *)dataString.data(), dataString.size());
 
         return data;
     }
 #endif
-    return ndn::ptr_lib::shared_ptr<Data>(nullptr);
+    return std::shared_ptr<Data>(nullptr);
 }
 
-ndn::ptr_lib::shared_ptr<Data> StorageEngineImpl::read(const Interest &interest)
+std::shared_ptr<Data> StorageEngineImpl::read(const Interest &interest)
 {
-    ndn::ptr_lib::shared_ptr<Data> data;
+    std::shared_ptr<Data> data;
+    // NOTE: Why isn't getCanBePrefix() a const function?
+    bool canBePrefix = const_cast<Interest&>(interest).getCanBePrefix();
 
     if (canBePrefix)
     {
         // TODO: implement prefix match data retrieval
         // extract by prefix match
         Name prefix = interest.getName(), keyName;
         auto it = db_->NewIterator(db_namespace::ReadOptions());
-        //db_namespace::Iterator *lastIt = nullptr;
-        // lastIt won't work because it changes with it.
-        boost::optional<std::string> lastKey = boost::none;
+        std::string key = "";
```

Comment:
good call on using `boost::optional`. I overlooked it. might change it back, if you want